src/main/webapp/app/components/tabs/ReviewHistoryTab.tsx

```diff
@@ -0,0 +1,89 @@
+import { GET_ALL_DRUGS_PAGE_SIZE } from 'app/config/constants/constants';
+import { FB_COLLECTION } from 'app/config/constants/firebase';
+import { HistoryCollection } from 'app/shared/model/firebase/firebase.model';
+import { downloadFile } from 'app/shared/util/file-utils';
+import { getAllGeneHistoryForDateRange, getHistoryEntryStrings } from 'app/shared/util/firebase/firebase-history-utils';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { IRootStore } from 'app/stores';
+import { onValue, ref } from 'firebase/database';
+import React, { useEffect, useState } from 'react';
+import { Button, Input, Label, Row, Spinner } from 'reactstrap';
+
+function ReviewHistoryTab({ firebaseDb, drugList, getDrugs }: StoreProps) {
+  const [historyCollection, setHistoryCollection] = useState<HistoryCollection>(null);
+  const [startDate, setStartDate] = useState<string>('');
+  const [endDate, setEndDate] = useState<string>('');
+
+  useEffect(() => {
+    const unsubscribe = onValue(ref(firebaseDb, FB_COLLECTION.HISTORY), snapshot => {
+      setHistoryCollection(snapshot.val());
+    });
+
+    return () => unsubscribe?.();
+  }, []);
+
+  useEffect(() => {
+    getDrugs({ page: 0, size: GET_ALL_DRUGS_PAGE_SIZE, sort: 'id,asc' });
+  }, []);
+
+  function handleDownload() {
+    const start = startDate ? new Date(startDate) : null;
+    const end = endDate ? new Date(endDate) : null;
+
+    const history = getAllGeneHistoryForDateRange(historyCollection, drugList, start, end);
+
+    const content: string[] = [];
+    if (history.gene.length > 0) {
+      const segments: string[] = [];
+      segments.push('### Gene\n```');
+      segments.push(...getHistoryEntryStrings(history.gene));
+      segments.push('```\n');
+      content.push(segments.join('\n'));
+    }
+    if (history.alteration.length > 0) {
+      const segments: string[] = [];
+      segments.push('### Alteration\n```');
+      segments.push(...getHistoryEntryStrings(history.alteration));
+      segments.push('```\n');
+      content.push(segments.join('\n'));
+    }
+    if (history.evidence.length > 0) {
+      const segments: string[] = [];
+      segments.push('### Evidence\n```');
+      segments.push(...getHistoryEntryStrings(history.evidence));
+      segments.push('```\n');
+      content.push(segments.join('\n'));
+    }
+
+    downloadFile('README.md', content.join('\n'));
```

Comment:
what happens if the download fails?

Reply:
Good point, will notify error in that case.